src/bidiMapper/mapper.ts

```diff
@@ -15,79 +15,85 @@
  */
 import { CommandProcessor } from './commandProcessor';
 
-import { CdpClient } from './utils/cdpClient';
+import { CdpClient, CdpTransport, connectCdp } from './utils/cdpClient';
 import { BidiServer } from './utils/bidiServer';
 import { ServerBinding } from './utils/iServer';
 
 import { log } from './utils/log';
 const logSystem = log('system');
 
 declare global {
-  interface Window extends GlobalObj {}
-
-  interface GlobalObj {
+  interface Window {
     //`window.cdp` is exposed by `Target.exposeDevToolsProtocol` from the server side.
     // https://chromedevtools.github.io/devtools-protocol/tot/Target/#method-exposeDevToolsProtocol
     cdp: {
-      send: (string) => void;
-      onmessage: (string) => void;
+      send: (message: string) => void;
+      onmessage: (message: string) => void;
     };
 
     // `window.sendBidiResponse` is exposed by `Runtime.addBinding` from the server side.
-    sendBidiResponse: (string) => void;
+    sendBidiResponse: (response: string) => void;
 
     // `window.onBidiMessage` is called via `Runtime.evaluate` from the server side.
-    onBidiMessage: (string) => void;
+    onBidiMessage: (message: string) => void;
 
     // `window.setSelfTargetId` is called via `Runtime.evaluate` from the server side.
-    setSelfTargetId: (string) => void;
+    setSelfTargetId: (targetId: string) => void;
   }
 }
 
-const globalObj = window as GlobalObj;
-
 // Initiate `setSelfTargetId` as soon as possible to prevent race condition.
 const _waitSelfTargetIdPromise = _waitSelfTargetId();
 
 (async () => {
   window.document.documentElement.innerHTML = `<h1>Bidi mapper runs here!</h1><h2>Don't close.</h2>`;
   window.document.title = 'BiDi Mapper';
 
-  const cdpServer = _createCdpServer();
+  const cdpClient = _createCdpClient();
   const bidiServer = _createBidiServer();
 
   // Needed to filter out info related to BiDi target.
   const selfTargetId = await _waitSelfTargetIdPromise;
 
   // Needed to get events about new targets.
-  await _prepareCdp(cdpServer);
+  await _prepareCdp(cdpClient);
 
-  CommandProcessor.run(cdpServer, bidiServer, selfTargetId);
+  CommandProcessor.run(cdpClient, bidiServer, selfTargetId);
 
   logSystem('launched');
 
   bidiServer.sendMessage('launched');
 })();
 
-function _createCdpServer() {
-  const cdpBinding = new ServerBinding(
-    (message) => {
-      globalObj.cdp.send(message);
-    },
-    (handler) => {
-      globalObj.cdp.onmessage = handler;
+function _createCdpClient() {
+  // A CdpTransport implementation that uses the window.cdp bindings
+  // injected by Target.exposeDevToolsProtocol.
+  class WindowGlobalCdpTransport implements CdpTransport {
```

Comment:
`WindowGlobalCdpTransport` -> `WindowCdpTransport `
```suggestion
  class WindowCdpTransport implements CdpTransport {
```